### PR TITLE
Provide config file for gitpod

### LINKED ISSUE #118: Setup a config file for `Gitpod`
Write a Gitpod config file to allow contributors to quickly setup a full-fleged coding environment for making contributing to the project easier and smoother experience.

### PR DESCRIPTION
**Pull request by neon-mmd**
## What does this PR do?

This PR provides a config file for Gitpod. 

## Why is this change important?

This change is essential to allow contributors to contribute to the project by quickly setting up a working Gitpod environment hassle-free.

## How to test this PR locally?

To test this PR, launch the project with Gitpod and see if everything gets setup as expected. 

Gitpod link: [https://gitpod.io/#https://github.com/neon-mmd/websurfx](https://gitpod.io/#https://github.com/neon-mmd/websurfx)

## Author's checklist

- [x] Add config file for Gitpod.
- [x] Bump the `Cargo.toml` version.

## Related issues

closes #118

### CODE REVIEW

**Review by alamin655 — APPROVED** [review]
Nice work! I've merged your PR.

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+image: ubuntu:latest 
+# Commands that will run on workspace start 
+tasks: 
+    - name: Setup, Install & Build 
+    - before: apt install cargo redis-server nodejs npm && cargo test
+    - init: cargo install cargo-watch 
+    - command: redis-server --port 8080 & cargo watch -q -w "." -x "run"
```
Comment (alamin655) [inline .gitpod.yml:8]:
```suggestion
      init: cargo install cargo-watch 
      command: redis-server --port 8080 & cargo watch -q -w "." -x "run" 
```

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - name: Setup, Install & Build 
+    - before: apt install cargo redis-server nodejs npm && cargo test
+    - init: cargo install cargo-watch 
+    - command: redis-server --port 8080 & cargo watch -q -w "." -x "run" 
+# Ports to expose on workspace startup 
+ports: 
+    - port: 8080 
+    - onOpen: open-preview 
+    - name: Website 
+    - description: Website Preview
```
Comment (alamin655) [inline .gitpod.yml:14]:
```suggestion
      onOpen: open-preview 
      name: Website 
      description: Website Preview
```

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - eamodio.gitlens
+github: 
+   prebuilds: 
+    - master: true 
+    - branches: true 
+    - pullRequests: true 
+    - pullRequestsFromForks: true 
+    - addCheck: true 
+    - addComment: false 
+    - addBadge: true
```
Comment (alamin655) [inline .gitpod.yml:45]:
```suggestion
      master: true 
      branches: true 
      pullRequests: true 
      pullRequestsFromForks: true 
      addCheck: true 
      addComment: false 
      addBadge: true
```

**Review by alamin655 — COMMENTED** [review]
Update the config according to the GitPod reference.

**Comment by neon-mmd** [comment]
Thanks for all the suggestions and taking the time to review this PR but I would suggest having all this changes in a seperate PR titled "Improve Gitpod config file" and also I would suggest not reverting this PR because I feel that would be better 🙂, what do you say??

**Comment by alamin655** [comment]
> Thanks for all the suggestions and taking the time to review this PR but I would suggest having all this changes in a seperate PR titled "Improve Gitpod config file" and also I would suggest not reverting this PR because I feel that would be better 🙂, what do you say??
> […](#)
> On June 28, 2023 10:06:45 AM UTC, alamin655 ***@***.***> wrote: @alamin655 commented on this pull request. Update the config according to the GitPod reference. -- Reply to this email directly or view it on GitHub: [#122 (review)](https://github.com/neon-mmd/websurfx/pull/122#pullrequestreview-1502745971) You are receiving this because you authored the thread. Message ID: ***@***.***>

Well, it's not an improvement, as the current config isn't working.

**Comment by neon-mmd** [comment]
> > Thanks for all the suggestions and taking the time to review this PR but I would suggest having all this changes in a seperate PR titled "Improve Gitpod config file" and also I would suggest not reverting this PR because I feel that would be better 🙂, what do you say??
> > […](#)
> > On June 28, 2023 10:06:45 AM UTC, alamin655 _**@**_._**> wrote: @alamin655 commented on this pull request. Update the config according to the GitPod reference. -- Reply to this email directly or view it on GitHub: [#122 (review)](https://github.com/neon-mmd/websurfx/pull/122#pullrequestreview-1502745971) You are receiving this because you authored the thread. Message ID: _**@**_.**_>
> 
> Well, it's not an improvement, as the current config isn't working.

But I feel it would be a better idea to have it in a separate PR. Because this PR has been already merged, so adding changes on top of it is not possible because of how GitHub works (shown in the image below), and also it makes sense why they would do so. So I would suggest making another PR fixing this PR instead of reverting it. Also, adding a new PR would add a lot of value because this will make reviewing the PR easier as well as this will add more collaboration, What do you say?? :slightly_smiling_face: 

image

**Pull request merged**